vendor/github.com/insolar/insconfig/configurator.go

```diff
@@ -0,0 +1,236 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package insconfig
+
+import (
+	goflag "flag"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	flag "github.com/spf13/pflag"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
+)
+
+// This should be implemented by local config struct
+type ConfigStruct interface {
+	GetConfig() interface{}
+}
+
+type Params struct {
+	// Your config
+	ConfigStruct ConfigStruct
+	// Prefix for environment variables
+	EnvPrefix string
+	// Custom viper decoding hooks
+	ViperHooks []mapstructure.DecodeHookFunc
+	// Should return config path
+	ConfigPathGetter ConfigPathGetter
+	// If set then return error on file not found
+	FileRequired bool
+}
+
+type ConfigPathGetter interface {
+	GetConfigPath() string
+}
+
+// Default behaviour adds "--config" flag and read path from it, custom flags should be created before
+type DefaultConfigPathGetter struct {
+	// For go flags compatibility
+	GoFlags *goflag.FlagSet
+	// For spf13/pflags compatibility
+	PFlags *flag.FlagSet
+}
+
+func (g DefaultConfigPathGetter) GetConfigPath() string {
+	if g.GoFlags != nil {
+		flag.CommandLine.AddGoFlagSet(g.GoFlags)
+	}
+	if g.PFlags != nil {
+		flag.CommandLine.AddFlagSet(g.PFlags)
+	}
+	configPath := flag.String("config", "", "path to config")
+	flag.Parse()
+	return *configPath
+}
+
+type insConfigurator struct {
+	params Params
+	viper  *viper.Viper
+}
+
+func NewInsConfigurator(params Params) insConfigurator {
+	return insConfigurator{
+		params: params,
+		viper:  viper.New(),
+	}
+}
+
+// Loads configuration from path and making checks
+func (i *insConfigurator) Load() (ConfigStruct, error) {
+	if i.params.EnvPrefix == "" {
+		return nil, errors.New("EnvPrefix should be defined")
+	}
+
+	configPath := i.params.ConfigPathGetter.GetConfigPath()
+	return i.load(configPath, i.params.FileRequired)
+}
+
+func (i *insConfigurator) load(path string, required bool) (ConfigStruct, error) {
+
+	i.viper.AutomaticEnv()
+	i.viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	i.viper.SetEnvPrefix(i.params.EnvPrefix)
+
+	i.viper.SetConfigFile(path)
+	if err := i.viper.ReadInConfig(); err != nil {
+		if required {
+			return nil, err
+		}
+		fmt.Printf("failed to load config from '%s'\n", path)
+	}
+	actual := i.params.ConfigStruct.GetConfig()
+	i.params.ViperHooks = append(i.params.ViperHooks, mapstructure.StringToTimeDurationHookFunc(), mapstructure.StringToSliceHookFunc(","))
+	err := i.viper.UnmarshalExact(actual, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
+		i.params.ViperHooks...,
+	)))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal config file into configuration structure")
```

Comment:
failed to unmarshal `the` config file into configuration structure

Reply:
I think, no